source/manual/howto-ssh-to-machines-in-aws.html.md

```diff
@@ -4,81 +4,132 @@ title: SSH into AWS machines
 section: AWS accounts
 layout: manual_layout
 parent: "/manual.html"
-last_reviewed_on: 2019-03-17
+last_reviewed_on: 2019-07-11
 review_in: 6 months
 ---
 
-This document explains how to SSH into machines in AWS, as it's markedly
-different than for Carrenza.
+This document explains how to SSH into machines in AWS, and what commands exist
+to navigate machines and applications quickly.
 
-In AWS, there are no static hostnames, so we can't have
-`backend-1.backend.integration` to SSH to like in Carrenza. EC2 instances have
-dynamically assigned IPs, which means hostnames like
-`ip-10-1-5-53.eu-west-1.compute.internal`. Each Puppeted instance has a "node
-class" (backend, frontend, ...), and the list of instances belonging to these
+![Visualisation of commands from dev to production machine](./images/govuk-cli-ssh-overview.png)
+
+In AWS, there are no static hostnames, so we can't have `backend-1.backend.integration`
+to SSH to like in Carrenza. EC2 instances have dynamically assigned IPs, which means
+hostnames like `ip-10-1-5-53.eu-west-1.compute.internal`. Each Puppeted instance has a
+"node class" (backend, frontend, ...) and the list of instances belonging to these
 classes is accessible via `govuk_node_list` when logged onto the environment.
 
-To help connecting to the environments, there is a wrapper tool called [`govukcli`](https://github.com/alphagov/govuk-aws/blob/master/tools/govukcli).
+To help connecting to the environments, there is a wrapper tool called
+[`govukcli`](https://github.com/alphagov/govuk-aws/blob/master/tools/govukcli). The tool
+uses the idea of "contexts", where a "context" is a specific environment. By setting a
+context, any subsequent `govukcli` commands will be tied to the environment you chose.
 
-## Install
+`govuk_node_list`, `govuk_app_console` and `govuk_app_dbconsole` are also referenced in
+this document. These common commands live in [govuk-puppet](https://github.com/alphagov/govuk-puppet).
```

Comment:
Where are `govuk_app_console ` and `govuk_app_dbconsole` documented? Can you add a link?

Reply:
Nowhere, AFAIK. This is the first attempt at documenting them (covered in the penultimate section of this page).

Only things I can think of linking out to would be [the file in the code](https://github.com/alphagov/govuk-puppet/blob/master/modules/govuk_scripts/files/usr/local/bin/govuk_app_dbconsole) or [the PR in which each was added](https://github.com/alphagov/govuk-puppet/pull/9050).

Reply:
Oh, you mean further down. I would remove this line. It's a bit confusing. I thought "this document" referred to https://github.com/alphagov/govuk-aws/blob/master/tools/govukcli

Reply:
Ahhh ok 👍